client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningGetResultsIT.java

```diff
@@ -145,11 +149,384 @@ private void addCategoriesIndexRequests(BulkRequest bulkRequest) {
+            assertThat(response.snapshots().get(1).getModelSizeStats().getModelBytes(), equalTo(51722L));
+            assertThat(response.snapshots().get(1).getModelSizeStats().getTotalByFieldCount(), equalTo(3L));
+            assertThat(response.snapshots().get(1).getModelSizeStats().getTotalOverFieldCount(), equalTo(0L));
+            assertThat(response.snapshots().get(1).getModelSizeStats().getTotalPartitionFieldCount(), equalTo(2L));
+            assertThat(response.snapshots().get(1).getModelSizeStats().getBucketAllocationFailuresCount(), equalTo(0L));
+            assertThat(response.snapshots().get(1).getModelSizeStats().getMemoryStatus(),
+                equalTo(ModelSizeStats.MemoryStatus.fromString("ok")));
+
+            assertThat(response.snapshots().get(0).getJobId(), equalTo(JOB_ID));
+            assertThat(response.snapshots().get(0).getSnapshotId(), equalTo("1541589919"));
+            assertThat(response.snapshots().get(0).getSnapshotDocCount(), equalTo(1));
+            assertThat(response.snapshots().get(0).getDescription(), equalTo("State persisted due to job close at" +
+                " 2018-11-07T11:25:19+0000"));
+            assertThat(response.snapshots().get(0).getSnapshotDocCount(), equalTo(1));
+            assertThat(response.snapshots().get(0).getTimestamp(), equalTo(new Date(1541589919000L)));
+            assertThat(response.snapshots().get(0).getLatestRecordTimeStamp(), equalTo(new Date(1519931700000L)));
+            assertThat(response.snapshots().get(0).getLatestResultTimeStamp(), equalTo(new Date(1519930800000L)));
+            assertThat(response.snapshots().get(0).getModelSizeStats().getJobId(), equalTo(JOB_ID));
+            assertThat(response.snapshots().get(0).getModelSizeStats().getModelBytes(), equalTo(51722L));
+            assertThat(response.snapshots().get(0).getModelSizeStats().getTotalByFieldCount(), equalTo(3L));
+            assertThat(response.snapshots().get(0).getModelSizeStats().getTotalOverFieldCount(), equalTo(0L));
+            assertThat(response.snapshots().get(0).getModelSizeStats().getTotalPartitionFieldCount(), equalTo(2L));
+            assertThat(response.snapshots().get(0).getModelSizeStats().getBucketAllocationFailuresCount(), equalTo(0L));
+            assertThat(response.snapshots().get(0).getModelSizeStats().getMemoryStatus(),
+                equalTo(ModelSizeStats.MemoryStatus.fromString("ok")));
+        }
+        {
+            GetModelSnapshotsRequest request = new GetModelSnapshotsRequest(JOB_ID);
+            request.setSort("timestamp");
+            request.setDesc(false);
+            request.setPageParams(new PageParams(0, 1));
+
+            GetModelSnapshotsResponse response = execute(request, machineLearningClient::getModelSnapshots,
+                machineLearningClient::getModelSnapshotsAsync);
+
+            assertThat(response.count(), equalTo(3L));
```

Comment:
Hmmm, I think you mean for this to be `1L`, this goes for similar tests below.

Reply:
Thanks for the review Ben - very thorough! The response count represents the total number of snapshots associated with the job hence is constant regardless of the for of a (valid) query. `3L` is correct in these instances.

Reply:
Interesting, as that is not how the other `AbstractResponse<T>` classes act. I suppose if the tests pass it is OK. 

Usually, `count()` represents the number of items returned (basically, the same size as the list itself).

Reply:
It works like the `total_hits` of elasticsearch. So it will return as many docs match the query despite of how many fit in a page.